src/jit/lower.cpp

```diff
@@ -4984,9 +4998,7 @@ GenTree* Lowering::LowerSignedDivOrMod(GenTree* node)
     GenTree* dividend = divMod->gtGetOp1();
     GenTree* divisor  = divMod->gtGetOp2();
 
-#ifdef _TARGET_XARCH_
-    if (!varTypeIsFloating(node->TypeGet()))
-#endif // _TARGET_XARCH_
+    if (varTypeIsIntegral(node->TypeGet()))
```

Comment:
The ifdef was probably intended to include the entire if, not just the condition. I don't think I enabled magic division for ARM64 when I wrote the code, it didn't have `GT_MULHI` back then. I see that now it's enabled for ARM64 but it's still not enabled for ARM.

Reply:
It is not enabled for arm because we do not allow integer MOD/DIV in Lower on this platform.

So, I do not see a reason why we need `target ifdef`, it works fine for XARCH DIV/MOD and for arm64 DIV nodes (MOD are not allowed in Lower).

Reply:
I don't see a reason why it should be target-specific either.